src/middleware/validateApiKey.ts

```diff
@@ -42,9 +43,9 @@ const findDeveloper = async (apiKey) => {
   // This logic attempts to find the developer document and update it
   // with the API token
   const developers = await Developer.find({});
-  developer = developers.find((dev) => compareSync(apiKey, dev.apiKey));
+  developer = developers.find((dev) => compareSync(apiKey as string, dev.apiKey))!;
```

Comment:
please remove the bang (`!`) at the end of this line since we check on line 47 if the developer exists or not